src/error.ts

```diff
@@ -32,3 +32,28 @@ export class FirebaseError extends Error {
     this.status = defaultTo(options.status, DEFAULT_STATUS);
   }
 }
+
+/**
+ * Checks if a FirebaseError is caused by attempting something
+ * that requires billing enabled while billing is not enabled.
+ */
+export function isBillingError(e: {
+  context?: {
+    body?: {
+      error?: {
+        details?: {
+          type: string;
+          reason?: string;
+          violations?: { type: string }[];
+        }[];
+      };
+    };
+  };
+}): boolean {
+  return !!e.context?.body?.error?.details?.find((d) => {
+    return (
+      d.violations?.find((v) => v.type === "serviceusage/billing-enabled") ||
+      d.reason == "UREQ_PROJECT_BILLING_NOT_FOUND"
```

Comment:
Added this check because the violation returned when failing to enable an API is not specific to billing issues - it looks like:
```
[ { type: 'googleapis.com',
    subject:
     '?error_code=390001&project=1747788017&services=cloudbuild.googleapis.com&services=containerregistry.googleapis.com' } ]
```